kasa/modules/ledmodule.py

```diff
@@ -0,0 +1,49 @@
+"""Module for base light effect module."""
+
+from __future__ import annotations
+
+from ..feature import Feature
+from ..module import Module
+
+
+class LedModule(Module):
+    """Base interface to represent a LED module."""
+
+    # This needs to implement abstract methods for typing to work with
+    # overload get_module(type[ModuleT]) -> ModuleT:
+    # https://discuss.python.org/t/add-abstracttype-to-the-typing-module/21996
+
+    def _initialize_features(self):
+        """Initialize features."""
+        device = self._device
+        self._add_feature(
+            Feature(
+                device=device,
+                container=self,
+                name="LED",
+                id="led",
+                icon="mdi:led-{state}",
```

Comment:
```suggestion
                icon="mdi:led",
```
The state handling was removed earlier, as homeassistant should use icon translations for that.

Reply:
Fixed